src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -155,3 +260,146 @@ def asset_download(**args):
+    """
+
+    # root_path = asset_root_manifest.root_path
+
+    # input_paths = []
+    # for root, dirs, files in os.walk(root_path):
+    #     if os.path.samefile(root, manifest):
+    #         dirs[:] = []
+    #         continue
+    #     for filename in files:
+    #         file_path = os.path.join(root, filename)
+    #         input_paths.append(file_path)
+
+    # return input_paths
+
+
+def get_manifest_changes(
+    asset_manager: S3AssetManager,
+    asset_root_manifest: AssetRootManifest,
+    manifest: str,
+    update: bool,
+) -> List[(tuple)]:
+    """
+    Gets the file paths in specified manifest if the contents of file have changed since its last snapshot.
+    """
+    manifest_dir_name = os.path.basename(manifest)
+    root_path = asset_root_manifest.root_path
+    input_paths: List[Path] = []
+
+    asset_manifest = asset_root_manifest.asset_manifest
+    if asset_manifest is None:
+        raise NonValidInputError("Manifest object not found, please check input manifest. ")
+
+    for base_manifest_path in asset_manifest.paths:
+        if base_manifest_path.path.startswith(manifest_dir_name):
+            # skip the manifest folder, or else every upload will need an update after a previous change
+            continue
+        input_paths.append(Path(root_path, base_manifest_path.path))
+
+    return get_file_changes(
+        asset_manager=asset_manager, input_paths=input_paths, root_path=root_path, update=update
+    )
+
+
+def get_file_changes(
+    asset_manager: S3AssetManager, input_paths: List[Path], root_path: str, update: bool
+) -> List[(tuple)]:
+    """
+    Checks a manifest file, compares it to specified root directory or manifest of files with the local hash cache.
+    Returns a list of tuples containing the file information, and its corresponding file status.
+    """
+    cache_config = config_file.get_cache_directory()
+
+    with HashCache(cache_config) as hash_cache:
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            futures = {
+                executor.submit(
+                    asset_manager._process_input_path,
+                    path=path,
+                    root_path=root_path,
+                    hash_cache=hash_cache,
+                    update=update,
+                ): path
+                for path in input_paths
+            }
+            new_or_modified_paths: List[(tuple)] = []
+            for future in concurrent.futures.as_completed(futures):
+                (file_status, _, manifestPath) = future.result()
+                if file_status is FileStatus.NEW or file_status is FileStatus.MODIFIED:
+                    new_or_modified_paths.append((file_status, manifestPath))
+
+            return new_or_modified_paths
+
+
+def update_manifest(manifest: str, new_or_modified_paths: List[(tuple)]) -> BaseAssetManifest:
```

Comment:
nit: typing for `new_or_modified_paths` is a bit weird. I don't think we need the round brackets around `tuple`.